### PR TITLE
fix(amplify-category-analytics): reverted the hyphen and updated tests

### LINKED ISSUE #3163: `amplify add analytics` successfully with dash in resource name but fail on `amplify push`
**Describe the bug**
When I run `amplify add analytics` and then specify a name with a dash, it accepts it. This should just fail with validation logic because when I go to `amplify push`, it fails with 
```
Template format error: Resource name analyticsanalytics-with-dash is non alphanumeric.
An error occured during the push operation: Template format error: Resource name analyticsanalytics-with-dash is non alphanumeric.
```
**Amplify CLI Version**
`4.12.0`

**To Reproduce**
Steps to reproduce the behavior or terminal output if applicable
1. `amplify init`, I chose iOS
2. `amplify add analytics`
3. Provide your pinpoint resource name: `analytics-with-dash`
4. `amplify push`

**Expected behavior**
Earlier client validation on pinpoint resource name, or provide a way for developer to update the resource name after the failed `amplify push.

**Screenshots**
image

### PR DESCRIPTION
**Pull request by ammarkarachi**
*Issue #, if available:*
fix #3163
*Description of changes:*

The hyphens are not allowed when pushing and had to revert it back

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-e2e-tests/src/__tests__/analytics.test.ts [hunk 1]
```diff
@@ -12,13 +12,15 @@ describe('amplify add analytics', () => {
 
   afterEach(async () => {
     await removeAnalytics(projRoot, {});
-    deleteProjectDir(projRoot);
+    await deleteProject(projRoot);
+    deleteProjectDir(projRoot), true;
```
Comment (kaustavghosh06) [inline packages/amplify-e2e-tests/src/__tests__/analytics.test.ts]: Why do we need this?

File: packages/amplify-e2e-tests/src/__tests__/analytics.test.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import { initJSProjectWithProfile } from '../init';
+import { initJSProjectWithProfile, amplifyPushUpdate, deleteProject } from '../init';
```
Comment (kaustavghosh06) [inline packages/amplify-e2e-tests/src/__tests__/analytics.test.ts]: Is this a part of this PR?

File: packages/amplify-e2e-tests/src/__tests__/analytics.test.ts (hunk 1 above)
Comment (ammarkarachi) [reply packages/amplify-e2e-tests/src/__tests__/analytics.test.ts]: Sorry I think I missed that will fix it now

File: packages/amplify-e2e-tests/src/__tests__/analytics.test.ts (hunk 2 above)
Comment (ammarkarachi) [reply packages/amplify-e2e-tests/src/__tests__/analytics.test.ts]: Yes, it is to clean up resources

File: packages/amplify-e2e-tests/src/__tests__/analytics.test.ts (hunk 1 above)
Comment (ammarkarachi) [reply packages/amplify-e2e-tests/src/__tests__/analytics.test.ts]: Nvm It should be there to clean the folders. I thought I had that in twice

File: packages/amplify-e2e-tests/src/__tests__/analytics.test.ts (hunk 2 above)
Comment (kaustavghosh06) [reply packages/amplify-e2e-tests/src/__tests__/analytics.test.ts]: That should be a part of another PR then..doesn't look like an analytics fix

Comment (ammarkarachi) [reply packages/amplify-e2e-tests/src/__tests__/analytics.test.ts]: It is a part of the test for Analytics

File: packages/amplify-e2e-tests/src/__tests__/analytics.test.ts (hunk 1 above)
Comment (kaustavghosh06) [reply packages/amplify-e2e-tests/src/__tests__/analytics.test.ts]: the deleteProjectDir() function is async?
deleteProjectDir(projRoot), true ; ? (What's the true for?)

File: packages/amplify-e2e-tests/src/__tests__/analytics.test.ts (hunk 2 above)
Comment (kaustavghosh06) [reply packages/amplify-e2e-tests/src/__tests__/analytics.test.ts]: Nah, deleted my comment.

File: packages/amplify-e2e-tests/src/__tests__/analytics.test.ts (hunk 1 above)
Comment (ammarkarachi) [reply packages/amplify-e2e-tests/src/__tests__/analytics.test.ts]: DeleteProjectDir cleans up the folders which is not async. DeleteProject cleans up the resources on the cloud. The true is to delete the deployment bucket

Comment (kaustavghosh06) [reply packages/amplify-e2e-tests/src/__tests__/analytics.test.ts]: true is outside the function call :|

**Pull request merged**